### PR TITLE
PLAT-63668: Revise Scrollable-related patches

### PR DESCRIPTION
**Pull request by ybsung**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

1. The patch for ENYO-5442: Needed to remove `setTimeout` if possible and created a new patch
2. The patches for ENYO-3984 and ENYO-5339:  We found some unnecessary code while creating several patches (`moonstone/Scrollable.componentDidUpdate()` is not needed anymore)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

1. The patch for ENYO-5442
: `this.isScrolledByJump` variable could not get a chance to be `false`. Then the bug in ENYO-5442 was reproduced. I fixed it to set `this.isScrolledByJump` variable to `false` when receiving a key event

2. The patches for ENYO-3984 and ENYO-5339
- Removed `moonstone/Scrollable.componentDidUpdate()` because we already covered in `moonstone/Scrollable.handleScrollerUpdate()`.
- Cleanup code.

I verified the bugs in ENYO-5442, ENYO-3984 and ENYO-5339 with this PR.

### Links
[//]: # (Related issues, references)

PLAT-63668

### Comments

Enact-DCO-1.0-Signed-off-by: YB Sung (yb.sung@lge.com)

### CODE REVIEW

**Comment by ybsung** [comment]
I'll update CHANGELOG.md after my previous several PRs are merged.

**Comment by ybsung** [comment]
I don't think that I need to update CHANGELOG.md because there is no behavior change in this PR.

File: packages/moonstone/Scrollable/Scrollable.js [hunk 1]
```diff
@@ -572,8 +568,9 @@ class ScrollableBase extends Component {
 
 	// Callback for scroller updates; calculate and, if needed, scroll to new position based on focused item.
 	handleScrollerUpdate = () => {
-		if (this.uiRef.scrollToInfo === null && this.childRef.nodeIndexToBeFocused == null && Spotlight.getPointerMode()) {
-			this.calculateAndScrollTo(Spotlight.getCurrent());
+		// For Scroller
+		if (this.uiRef.scrollToInfo === null && !this.hasOwnProperty(this.childRef.restoreLastFocused) && Spotlight.getPointerMode()) {
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/Scrollable.js]: Do we need this condition?
`!this.hasOwnProperty..`

**Comment by MikyungKim** [comment]
LGTM

**Pull request merged**